agent/provider/src/rules.rs

```diff
@@ -89,6 +89,52 @@ impl RulesManager {
         }
     }
 
+    pub fn set_audited_mode(&self, cert_id: String, mode: Mode) -> Result<()> {
+        let cert_id = {
+            let certs: Vec<Cert> = self
+                .keystore
+                .list()
+                .into_iter()
+                .filter(|cert| cert.id().starts_with(&cert_id))
+                .collect();
+
+            if certs.is_empty() {
+                bail!(
+                    "Setting Audited-Payload mode {mode} failed: No cert id: {cert_id} found in keystore"
+                );
+            } else if certs.len() > 1 {
+                bail!(
+                    "Setting Audited-Payload mode {mode} failed: Cert id: {cert_id} isn't unique"
+                );
+            } else {
+                let cert = &certs[0];
+                match cert {
+                    Cert::X509(_) => bail!(
+                        "Failed to set audited-payload mode for certificate {cert_id}. Audited-Payload mode can be set only for Golem certificate."
+                    ),
+                    Cert::Golem { id, .. } => id.clone(),
+                }
```

Comment:
Whis is wrong: X509 certs are used in Audited-Payload rule.

Reply:
Done, but it needs a test.